src/transformers/models/mask2former/modeling_mask2former.py

```diff
@@ -0,0 +1,2416 @@
+
+        Returns:
+            `Mask2FormerUniversalSegmentationOutput`
+
+        Examples:
+        ```python
+        >>> from transformers import Mask2FormerImageProcessor, Mask2FormerForUniversalSegmentation
+        >>> from PIL import Image
+        >>> import requests
+        >>> import torch
+
+        >>> # Load Mask2Former trained on ADE20K instance segmentation dataset
+        >>> image_processor = Mask2FormerImageProcessor.from_pretrained("facebook/mask2former-instance-swin-small-ade")
+        >>> model = Mask2FormerForUniversalSegmentation.from_pretrained("facebook/mask2former-instance-swin-small-ade")
+
+        >>> url = (
+        ...     "https://huggingface.co/datasets/hf-internal-testing/fixtures_ade20k/resolve/main/ADE_val_00000001.jpg"
+        ... )
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+        >>> inputs = image_processor(image, return_tensors="pt")
+
+        >>> with torch.no_grad():
+        ...     outputs = model(**inputs)
+
+        >>> # Model predicts class_queries_logits of shape `(batch_size, num_queries)`
+        >>> # and masks_queries_logits of shape `(batch_size, num_queries, height, width)`
+        >>> class_queries_logits = outputs.class_queries_logits
+        >>> masks_queries_logits = outputs.masks_queries_logits
+
+        >>> # Perform post-processing to get semantic, instance or panoptic segmentation maps
+        >>> pred_semantic_map = feature_extractor.post_process_instance_segmentation(
+        ...     outputs, target_sizes=[image.size[::-1]]
+        ... )[0]
+        >>> pred_instance_map = feature_extractor.post_process_instance_segmentation(
+        ...     outputs, target_sizes=[image.size[::-1]]
+        ... )[0]["segmentation"]
+        >>> pred_panoptic_map = feature_extractor.post_process_panoptic_segmentation(
+        ...     outputs, target_sizes=[image.size[::-1]]
+        ... )[0]["segmentation"]
+        ```
+        """
+
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        outputs: Mask2FormerModelOutput = self.model(
+            pixel_values=pixel_values,
+            pixel_mask=pixel_mask,
+            output_hidden_states=output_hidden_states or self.config.general_config["use_auxiliary_loss"],
+            output_attentions=output_attentions,
+            return_dict=True,
+        )
+
+        loss, loss_dict, auxiliary_predictions = None, None, None
+
+        class_queries_logits = outputs.transformer_decoder_class_predictions
+        masks_queries_logits = outputs.transformer_decoder_mask_predictions
+        auxiliary_predictions = outputs.transformer_decoder_auxiliary_predictions
```

Comment:
It seems the logits are computed inside the base model, rather than with heads on top?

Normally to conform to the design of other models in the library, the head is added in `xxxForUniversalSegmentation`